src/Yasumi/Provider/Argentina.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+declare(strict_types=1);
+/*
+ * This file is part of the Yasumi package.
+ *
+ * Copyright (c) 2015 - 2021 AzuyaLabs
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @author Nader Safadi <nader@gizra.com>
+ */
+
+namespace Yasumi\Provider;
+
+use DateInterval;
+use DateTime;
+use Yasumi\Exception\InvalidDateException;
+use Yasumi\Exception\UnknownLocaleException;
+use Yasumi\Holiday;
+
+/**
+ * Provider for all holidays in Argentina.
+ */
+class Argentina extends AbstractProvider
+{
+    use CommonHolidays;
+    use ChristianHolidays;
+
+    /**
+     * Code to identify this Holiday Provider. Typically, this is the ISO3166
+     * code corresponding to the respective country or sub-region.
+     */
+    public const ID = 'AR';
+
+    /**
+     * Initialize holidays for Argentina.
+     *
+     * @throws InvalidDateException
+     * @throws \InvalidArgumentException
+     * @throws UnknownLocaleException
+     * @throws \Exception
+     */
+    public function initialize(): void
+    {
+        $this->timezone = 'America/Argentina/Buenos_Aires';
+
+        // Add common holidays
+        $this->addHoliday($this->newYearsDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->internationalWorkersDay($this->year, $this->timezone, $this->locale));
+
+        // Add Christian holidays
+        $this->addHoliday($this->christmasDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->easter($this->year, $this->timezone, $this->locale, Holiday::TYPE_OBSERVANCE));
+        $this->addHoliday($this->goodFriday($this->year, $this->timezone, $this->locale, Holiday::TYPE_OBSERVANCE));
+
+        /*
+         * Carnaval
+         *
+         * Carnaval is the biggest popular festival of country. The festival
+         * happens on Day 48 and 47 before Easter.
+         *
+         * @link https://en.wikipedia.org/wiki/Brazilian_Carnival
+         */
+        if ($this->year >= 1700) {
+            $easter = $this->calculateEaster($this->year, $this->timezone);
+
+            $carnavalMonday = clone $easter;
+            $this->addHoliday(new Holiday(
+              'carnavalMonday',
+              [
+                'en' => 'Carnival Monday',
+                'es' => 'Lunes de Carnaval',
+              ],
+              $carnavalMonday->sub(new DateInterval('P48D')),
+              $this->locale,
+              Holiday::TYPE_OBSERVANCE
+            ));
+
+            $carnavalTuesday = clone $easter;
+            $this->addHoliday(new Holiday(
+              'carnavalTuesday',
+              [
+                'en' => 'Carnival Tuesday',
+                'es' => 'Martes de Carnaval',
+              ],
+              $carnavalTuesday->sub(new DateInterval('P47D')),
```

Comment:
The sub method can return `false` on failure, whilst the constructor 3rd parameter can only takes a `DateTimeInterface` object. It would be better to handle this situation.